modules/olm-adding-pullsecret-catalogsource.adoc

```diff
@@ -0,0 +1,46 @@
+// Module included in the following assemblies:
+//
+// * operators/admin/olm-restricted-networks.adoc
+// * operators/admin/managing-custom-catalogs.adoc
+
+[id="olm-adding-pullsecret-catalogsource_{context}"]
+= Adding pull secrets to catalog sources
+
+If an index image or catalog image is hosted on an authenticated container image registry, also known as a private registry, you can create a pull secret that contains the authentication credentials for the registry. By referencing the secret in a catalog source, the required images can be pulled for use in OperatorHub.
+
+.Prerequisites
+
+* An index image or catalog image built and pushed to a private registry.
+
+.Procedure
+
+. Create a secret that contains the authentication credentials for the private registry:
++
+[source,terminal]
+----
+$ oc create secret generic <secret_name> \
+    --from-file=.dockerconfigjson=/root/.docker/config.json \
+    --type=kubernetes.io/dockerconfigjson
+----
+
+. Create or update an existing `CatalogSource` object to reference your secret:
++
+[source,yaml]
+----
+apiVersion: operators.coreos.com/v1alpha1
+kind: CatalogSource
+metadata:
+  name: my-operator-catalog
+  namespace: openshift-marketplace
+spec:
+  sourceType: grpc
+  secrets: <1>
+  - "<secret_name>"
+  image: <registry>:<port>/<namespace>/<image>:<tag>
+  displayName: My Operator Catalog
+  publisher: <publisher_name>
+  updateStrategy:
+    registryPoll:
+      interval: 30m
+----
+<1> Add a `spec.secrets` section and specify the secret.
```

Comment:
It's better to add a `note` at the end: if the subscribed operator uses the private image, it will be failed to install. Because this `secret` will not be used for the Deployment. If the users want to this operator can be installed successfully. They have to add the auth info to the `pull-secret` of the `openshift-config` project. Related steps:
```console
$ oc extract secret/pull-secret -n openshift-config  --confirm
.dockerconfigjson
$ cat .dockerconfigjson | jq --compact-output '.auths["<registry>:<port>/<namespace>/"] |= . + {"auth":"xxx"}' > new_dockerconfigjson
$ oc set data secret/pull-secret -n openshift-config  --from-file=.dockerconfigjson=new_dockerconfigjson
secret/pull-secret data updated
```
Details: https://bugzilla.redhat.com/show_bug.cgi?id=1909992#c7